include/nanoeigenpy/decompositions/minres.hpp

```diff
@@ -9,26 +9,145 @@
 namespace nanoeigenpy {
 namespace nb = nanobind;
 
-template <typename MatrixType, typename MatrixOrVector>
-MatrixOrVector solve(const Eigen::MINRES<MatrixType> &c,
-                     const MatrixOrVector &vec) {
-  return c.solve(vec);
-}
-
-template <typename MatrixType, typename MatrixOrVector1,
-          typename MatrixOrVector2>
-MatrixOrVector1 solveWithGuess(const Eigen::MINRES<MatrixType> &c,
-                               const MatrixOrVector1 &b,
-                               const MatrixOrVector2 &guess) {
-  return c.solveWithGuess(b, guess);
-}
+struct IterativeSolverBaseVisitor
+    : nb::def_visitor<IterativeSolverBaseVisitor> {
+  template <typename _Solver, typename... Ts>
+  void execute(nb::class_<_Solver, Ts...> &cl) {
```

Comment:
Add a check that `Solver` is a subclass of `IterativeSolverBase<Solver>`, to check the concept is satisfied